internal/graph/resolve_check_request.go

```diff
@@ -39,13 +40,49 @@ type ResolveCheckRequestMetadata struct {
 	WasThrottled *atomic.Bool
 }
 
+type ResolveCheckRequestParams struct {
+	StoreID               string
+	TupleKey              *openfgav1.TupleKey
+	ContextualTuples      *openfgav1.ContextualTupleKeys
+	Context               *structpb.Struct
+	Consistency           openfgav1.ConsistencyPreference
+	CacheInvalidationTime time.Time
+	AuthorizationModelID  string
+}
+
 func NewCheckRequestMetadata() *ResolveCheckRequestMetadata {
 	return &ResolveCheckRequestMetadata{
 		DispatchCounter: new(atomic.Uint32),
 		WasThrottled:    new(atomic.Bool),
 	}
 }
 
+func NewResolveCheckRequest(
+	params ResolveCheckRequestParams,
+) (*ResolveCheckRequest, error) {
```

Comment:
Could we add unit tests for this function?
- storeId, modelID and tuplekey and InvariantCacheKey cannot be empty